crypto/evp/e_aes_cbc_hmac_sha1.c

```diff
@@ -859,6 +859,8 @@ static int aesni_cbc_hmac_sha1_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg,
                 key->payload_length = len;
                 if ((key->aux.tls_ver =
                      p[arg - 4] << 8 | p[arg - 3]) >= TLS1_1_VERSION) {
+                    if (len < AES_BLOCK_SIZE)
+                        return 0;
```

Comment:
I wonder what the correct return value should be? There seems to be confusion between the different ctrl functions. A few lines above here we return -1 on error. An equivalent very similar check in e_aes.c returns 0 though.